### DISCUSSION TITLE
How to load a texture from memory to use with image decorator ?

### DISCUSSION THREAD

**Discussion by Lefraudeur** [mikke89/RmlUi#486] [Q&A] [answered]
How to load a texture from memory to use with image decorator ?
I've tried with context->LoadDocumentFromMemory but it doesn't work

**Comment by barotto** [comment, marked as answer]
You can do that by adding a load from memory mechanism yourself in your implementation of `RenderInterface::LoadTexture(TextureHandle& texture_handle, Vector2i& texture_dimensions, const String& source)`.
You just need to distinguish between the texture types by the `source` parameter and then load the correct data.
Not sure if it's the best way to do it, but it's what I did myself.

**Comment by Lefraudeur** [comment]
Thank you ! it works well, I'm using premade backend for win32 and gl2 so I haven't thought about this.
But now I have a read access violation when I exit the program I don't know why.

**Reply by Lefraudeur** [reply]
nvm I found out why 
Rml::Factory::InstanceStyleSheetString returned a Rml::SharedPtr that I forgot to delete, so the texture wasn't unloading properly I guess

**Comment by plong009** [comment]
here is another way, use SetFileInterface() to hook LoadTexture(Rml::Vector2i& texture_dimensions, const Rml::String& source).
use your own FileInterface to check the "source" argument, if it has memory data flag, then feed the memory to render.
